include/avl.h

```diff
@@ -21,32 +21,75 @@
 
 #include "common.h"
 
+
 #pragma once
 
 typedef struct avl {
+	int keyNode;
```

Comment:
What value does `keyNode` introduce? In other words, what are you doing new with `keyNode` that wasn't done with value?

Reply:
key node is old value.
now i specializated node information of avl tree, now there are keyNode:data
(like dictionary). 
For example search now browse the tree in search of keyNode (old val) and it retrun data parameter if it is found.
Another example is insert: now insert a node with one keyNode (and the avl is ordinate respect him) and every node have her data

i did this for my application, and also because my theachers teached this is more generic Tree node. (I also thinked to add void *pointer for data and make it more free )